modules/generative-openai/additional/generate/generate_result.go

```diff
@@ -76,22 +77,30 @@ func (p *GenerateProvider) generatePerSearchResult(ctx context.Context, in []sea
 	return in, nil
 }
 
-func (p *GenerateProvider) generateForAllSearchResults(ctx context.Context, in []search.Result, task string, cfg moduletools.ClassConfig) ([]search.Result, error) {
+func (p *GenerateProvider) generateForAllSearchResults(ctx context.Context, in []search.Result, task string, properties []string, cfg moduletools.ClassConfig) ([]search.Result, error) {
 	var propertiesForAllDocs []map[string]string
 	for _, res := range in {
-		propertiesForAllDocs = append(propertiesForAllDocs, p.getTextProperties(res))
+		propertiesForAllDocs = append(propertiesForAllDocs, p.getTextProperties(res, properties))
 	}
 	generateResult, err := p.client.GenerateAllResults(ctx, propertiesForAllDocs, task, cfg)
 	p.setCombinedResult(in, 0, generateResult, err)
 	return in, nil
 }
 
-func (p *GenerateProvider) getTextProperties(result search.Result) map[string]string {
+func (p *GenerateProvider) getTextProperties(result search.Result, properties []string) map[string]string {
 	textProperties := map[string]string{}
 	schema := result.Object().Properties.(map[string]interface{})
 	for property, value := range schema {
-		if valueString, ok := value.(string); ok {
-			textProperties[property] = valueString
+		if properties != nil {
```

Comment:
I would change this condition to `if len(properties) > 0 {` and remove from `containsProperty` method `if len(properties) == 0 {` check